src/compiler/compile.ts

```diff
@@ -69,6 +69,19 @@ export function compile(
         trackInfos = getRelativeTrackInfo(specCopy, theme).trackInfos;
     }
 
+    // publish the fixed spec
+    // used for alt-gosling
+    try {
+        if (PubSub) {
+            PubSub.publish('specTraversed', {
```

Comment:
Regarding the name of the event, how about `specProcessed` instead? I think users might not understand what the traverse means.

Reply:
That sounds like a good idea! I indeed was fidgetting around with a good name, specProcessed is more intuitive

Reply:
Sorry, bikeshedding, maybe `specResolved`?

Reply:
I like `specResolved`, which I think gives a better impression that the spec is not converted into another format, but rather the spec is slightly modified to resolve things!

Reply:
Wouldn't resolved give the idea that the spec is fully changed into the 'final form' that it gets for gosling? Because that isn't the case here, it is just partially processed. I'm not sure what the convention is, but I'm afraid resolved gives off the wrong idea?

Reply:
Happy to change it though

Reply:
Ah, I had the opposite idea! At this point, I don't have a strong preference between the two. Let's leave it as is and merge this when you think is ready.

Reply:
> Because that isn't the case here, it is just partially processed.

Sorry to dwell on this. What would we name an event that emitted a "fully processed" spec? Perhaps naming the event something more specific would be useful. How has the input been modified at this point in the program?

Reply:
> How has the input been modified at this point in the program?

The input spec is modified to have default values filled in, and some of the users' incorrectly-specified properties are fixed. Some parts of the spec are not touched by the program (e.g., when values are already specified by users, so no need to add default values), which I think is what @thomcsmits was referring to by `"it is just partially processed"`.

Reply:
Hm, we don't emit a fully processed spec ("fully processed" as in the final state that it becomes in Gosling), but that might happen in the future

Reply:
I don't have truely strong opinions on this, so I'm happy to switch it to specResolved still

Reply:
https://github.com/gosling-lang/gosling.js/pull/985 (as this PR is already merged)

Reply:
I don't really have a strong opinion about "resolved" vs "processed", but think this could be a good exercise to be explicit about the state of the gosling spec when it is emitted. "Resolved" and "processed" convey that completion of some transformations/resolution, which is not the case. I have left a longer comment in #985